src/lib/InstantSearch.ts

```diff
@@ -243,6 +244,39 @@ See ${createDocumentationLink({
       const routerOptions = typeof routing === 'boolean' ? undefined : routing;
       this.EXPERIMENTAL_use(createRouter(routerOptions));
     }
+
+    if (this.insightsClient) {
+      this.setupUserTokenUpdater();
+    }
+  }
+
+  /**
+   * Updates userToken automatically.
+   *
+   * When insightsClient is given, this method sets up an event listener
+   * so that every time userToken is set to insightsClient, it can set it to
+   * searchParameter, too.
+   */
+  private setupUserTokenUpdater() {
+    const configUserToken = connectConfigure(
+      ({ refineLater }, isFirstRendering) => {
+        if (!isFirstRendering) {
+          return;
+        }
+
+        (this.insightsClient as Function)(
+          'onSetUserToken',
+          userToken => {
+            console.log('setting userToken!', userToken);
+            refineLater({ userToken });
```

Comment:
I actually feel lie this should be done with `helper.setState` or `setSearchParameter` instead of the overhead of configure (and therefore also avoids us creating this new refineLater option :)

Reply:
Yeah it makes sense. Now that we have only one usage of `refineLater` here, we could just not add the option to the connector and later consider adding one if needed somewhere else.